lib/internal/streams/web.js

```diff
@@ -0,0 +1,4632 @@
+  constructor(controller, view) {
+    this[kType] = 'ReadableStreamBYOBRequest';
+    this[kState] = {
+      controller,
+      view,
+    };
+  }
+}
+
+class DefaultReadRequest {
+  constructor() {
+    this[kState] = createDeferredPromise();
+  }
+
+  [kChunk](value) {
+    this[kState].resolve?.({ value, done: false });
+  }
+
+  [kClose]() {
+    this[kState].resolve?.({ value: undefined, done: true });
+  }
+
+  [kError](error) {
+    this[kState].reject?.(error);
+  }
+
+  get promise() { return this[kState].promise; }
+}
+
+class ReadIntoRequest {
+  constructor() {
+    this[kState] = createDeferredPromise();
+  }
+
+  [kChunk](value) {
+    this[kState].resolve?.({ value, done: false });
+  }
+
+  [kClose](value) {
+    this[kState].resolve?.({ value, done: true });
+  }
+
+  [kError](error) {
+    this[kState].reject?.(error);
+  }
+
+  get promise() { return this[kState].promise; }
+}
+
+class ReadableStreamDefaultReader {
+  [kType] = 'ReadableStreamDefaultReader';
+
+  /**
+   * @param {ReadableStream} stream
+   */
+  constructor(stream) {
+    if (!isReadableStream(stream))
+      throw new ERR_INVALID_ARG_TYPE('stream', 'ReadableStream', stream);
+    this[kState] = {
+      readRequests: [],
+      stream: undefined,
+      close: {
+        promise: undefined,
+        resolve: undefined,
+        reject: undefined,
+      },
+    };
+    setupReadableStreamDefaultReader(this, stream);
+  }
+
+  /**
+   * @returns {Promise<{
+   *   value : any,
+   *   done : boolean
+   * }>}
+   */
+  read() {
+    if (!isReadableStreamDefaultReader(this))
+      throw new ERR_INVALID_THIS('ReadableStreamDefaultReader');
+    if (this[kState].stream === undefined) {
+      return PromiseReject(
+        new ERR_INVALID_STATE.TypeError(
+          'The reader is not attached to a stream'));
+    }
+    const readRequest = new DefaultReadRequest();
+    readableStreamDefaultReaderRead(this, readRequest);
+    return readRequest.promise;
+  }
+
+  releaseLock() {
+    if (!isReadableStreamDefaultReader(this))
+      throw new ERR_INVALID_THIS('ReadableStreamDefaultReader');
+    if (this[kState].stream === undefined)
+      return;
+    if (this[kState].readRequests.length) {
+      throw new ERR_INVALID_STATE.TypeError(
+        'Cannot release with pending read requests');
+    }
+    readableStreamReaderGenericRelease(this);
+  }
+
+  /**
+   * @readonly
+   * @type {Promise<void>}
+   */
+  get closed() {
+    if (!isReadableStreamDefaultReader(this))
+      throw new ERR_INVALID_THIS('ReadableStreamDefaultReader');
+    return this[kState].close.promise;
+  }
+
+  /**
+   * @param {any} reason
+   * @returns {Promise<void>}
+   */
+  cancel(reason) {
+    if (!isReadableStreamDefaultReader(this))
+      throw new ERR_INVALID_THIS('ReadableStreamDefaultReader');
+    if (this[kState].stream === undefined) {
+      return PromiseReject(new ERR_INVALID_STATE.TypeError(
+        'The reader is not attached to a stream'));
+    }
```

Comment:
```suggestion
  async cancel(reason) {
    if (!isReadableStreamDefaultReader(this))
      throw new ERR_INVALID_THIS('ReadableStreamDefaultReader');
    if (this[kState].stream === undefined) {
      throw new ERR_INVALID_STATE.TypeError(
        'The reader is not attached to a stream');
    }
```
same as above.

there's a bunch of these, so i won't comment on the others - ideally they'd all be async, but if the web spec requires they not be, then that would be unfortunate.